rust/main/agents/relayer/src/msg/pending_message.rs

```diff
@@ -510,13 +510,20 @@ impl PendingMessage {
         ctx: Arc<MessageContext>,
         app_context: Option<String>,
     ) -> Self {
-        let mut pm = Self::new(
-            message,
-            ctx,
-            // Since we don't persist the message status for now, assume it's the first attempt
-            PendingOperationStatus::FirstPrepareAttempt,
-            app_context,
-        );
+        // Attempt to fetch status about message from database
+        let mut pm = match ctx.origin_db.retrieve_status_by_message_id(&message.id()) {
```

Comment:
I wonder if it worth to refactor this method: retrieve all the required bits of data from database and construct the instance of `PendingMessage` at the end?